### ISSUE TITLE
Compiling doesn't change learning rate of reloaded model. Instead changes learning-rate of optimizer

### ISSUE THREAD

**Issue by kindaTall** [tensorflow/tensorflow#70050]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.15.0

### Custom code

Yes

### OS platform and distribution

Google Colab

### Python version

3.10.12

### Current behavior?

I expected the compiled model to have the learning rate I was trying to set. Instead it kept the original learning rate.
When repeated twice, this didn't occure.

### Standalone code to reproduce the issue

[https://colab.research.google.com/drive/1JNQmWZc9OYa39h1xfZaz2Fa8CDhte94n?usp=sharing](https://colab.research.google.com/drive/1JNQmWZc9OYa39h1xfZaz2Fa8CDhte94n?usp=sharing) 
```shell


import sys

from pathlib import Path

import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense


def create_model():
    model = Sequential([
        Dense(64, activation='relu', input_shape=(10,)),
        Dense(1, activation='sigmoid')
    ])
    return model

def compile_model(model, learning_rate):
    model.compile(optimizer=tf.keras.optimizers.Adam(learning_rate=learning_rate),
                  loss='binary_crossentropy')


def save_model_(model, log_dir, struct_only=False):
    log_dir = Path(log_dir)
    log_dir.mkdir(parents=True, exist_ok=True)
    with open(log_dir / 'model.json', 'w') as f:
        f.write(model.to_json())
    model.save_weights(log_dir / 'weights/weights')


def load_model_(log_dir):
    log_dir = Path(log_dir)
    with open(log_dir / 'model.json', 'r') as f:
        model_json = f.read()
    model = tf.keras.models.model_from_json(model_json)
    model.load_weights(log_dir / 'weights/weights')
    return model


if __name__ == "__main__":
    print(f"TensorFlow version: {tf.__version__}")
    print(f"Python version: {sys.version}")

    model = create_model()
    initial_learning_rate = 0.001
    compile_model(model, initial_learning_rate)

    model_path = 'my_model'
    save_model_(model, model_path)
    loaded_model = load_model_(model_path)

    new_learning_rate = 0.0001
    compile_model(loaded_model, new_learning_rate)
    print(f"Optimizer's learning rate of the loaded model: {loaded_model.optimizer.learning_rate.numpy():e}")
    # 1e-3
    compile_model(loaded_model, new_learning_rate)
    print(f"Optimizer's learning rate of the loaded model: {loaded_model.optimizer.learning_rate.numpy():e}")
    # 1e-4
```

### Relevant log output

```shell
TensorFlow version: 2.15.0
Python version: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Optimizer's learning rate of the loaded model: 1.000000e-03
Optimizer's learning rate of the loaded model: 1.000000e-04
```

**Comment by tilakrayal** [comment]
@kindaTall,
Thank you for the issue. We also observed the same behavior. Could you please allow some time to deep dive and investigate the same behavior. Also could you please try and let  me know if you are facing the same in the keras3.0 version, Tensorflow v2.16. Thank you!

**Comment by kindaTall** [comment]
This seems to be indeed fixed by 2.16

```Console
TensorFlow version: 2.16.1
Python version: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Optimizer's learning rate of the loaded model: 1.000000e-04
Optimizer's learning rate of the loaded model: 1.000000e-04
```

Too bad for me, I am currently restricted to a Windows machine, which doesn't seem to have GPU support post 2.10.

**Comment by tilakrayal** [comment]
@kindaTall,
Glad the issue is resolved in tensorflow 2.16.1. Could you please feel free to move this issue to closed status and TensorFlow 2.10 was the last TensorFlow release that supported GPU on native-Windows. Starting with TensorFlow 2.11, you will need to install [TensorFlow in WSL2](https://tensorflow.org/install/pip#windows-wsl2), or install tensorflow or tensorflow-cpu and, optionally, try the [TensorFlow-DirectML-Plugin](https://github.com/microsoft/tensorflow-directml-plugin#tensorflow-directml-plugin-)

Thank you!

**Comment by kindaTall** [comment]
Thank you!
